modules/renter/proto/negotiate.go

```diff
@@ -102,12 +101,12 @@ func verifyRecentRevision(conn net.Conn, contract modules.RenterContract) error
 	if err := encoding.ReadObject(conn, &hostSignatures, 2048); err != nil {
 		return errors.New("couldn't read host signatures: " + err.Error())
 	}
-	// check that revision number matches; if it does, do a more thorough
-	// check by comparing unlock hashes.
-	if lastRevision.NewRevisionNumber != contract.LastRevision.NewRevisionNumber {
-		return fmt.Errorf("our revision number (%v) does not match the host's (%v)", contract.LastRevision.NewRevisionNumber, lastRevision.NewRevisionNumber)
-	} else if lastRevision.UnlockConditions.UnlockHash() != contract.LastRevision.UnlockConditions.UnlockHash() {
+	// Check that the unlock hashes match; if they do not, something is
+	// seriously wrong. Otherwise, check that the revision numbers match.
+	if lastRevision.UnlockConditions.UnlockHash() != contract.LastRevision.UnlockConditions.UnlockHash() {
 		return errors.New("unlock conditions do not match")
+	} else if lastRevision.NewRevisionNumber != contract.LastRevision.NewRevisionNumber {
+		return &recentRevisionError{contract.LastRevision.NewRevisionNumber, lastRevision.NewRevisionNumber}
 	}
```

Comment:
it will. After `recentRevisionError` is returned, `verifyRecentRevision` will be called again with the other revision, and the renter will only proceed if it returns `nil`.